ReinforcementDrawing/structure_drawing.py

```diff
@@ -0,0 +1,606 @@
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "Reinforcement Drawing"
+__author__ = "Suraj"
+__url__ = "https://www.freecadweb.org"
+
+
+from xml.etree import ElementTree
+
+import FreeCAD
+import Part
+import Draft
+import DraftGeomUtils
+import DraftVecUtils
+import WorkingPlane
+
+from SVGfunc import (
+    getSVGRootElement,
+    getPointSVG,
+    isPointInSVG,
+    getLineSVG,
+    isLineInSVG,
+    getSVGTextElement,
+    getSVGPathElement,
+    getArrowMarkerElement,
+)
+from .config import SVG_POINT_DIA_FACTOR
+
+
+def getRebarsSpanAxis(rebar):
+    if (
+        Draft.getType(rebar.Base) == "Wire"
+    ):  # Draft Wires can have "wrong" placement
+        axis = DraftGeomUtils.getNormal(rebar.Base.Shape)
+    else:
+        axis = rebar.Base.Placement.Rotation.multVec(FreeCAD.Vector(0, 0, -1))
+    if hasattr(rebar, "Direction"):
+        if not DraftVecUtils.isNull(rebar.Direction):
+            axis = FreeCAD.Vector(rebar.Direction)
+            axis.normalize()
+    return axis
+
+
+def getProjectionToSVGPlane(vec, plane):
+    nx = DraftVecUtils.project(vec, plane.u)
+    lx = nx.Length
+    if abs(nx.getAngle(plane.u)) > 0.1:
+        lx = -lx
+    ny = DraftVecUtils.project(vec, plane.v)
+    ly = ny.Length
+    if abs(ny.getAngle(plane.v)) > 0.1:
+        ly = -ly
+    # if techdraw: buggy - we now simply do it at the end
+    #    ly = -ly
+    return FreeCAD.Vector(lx, ly, 0)
+
+
+def getStructureMinMaxPoints(structure, view_plane):
+    min_x = 9999999
+    min_y = 9999999
+    max_x = -9999999
+    max_y = -9999999
```

Comment:
please explain me its purpose and logic.

Reply:
Hi @amrit3701 
The values here are initialized: min_{}  very large and max_{} very small.
And we are finding min and max values of x and y to get height and width of svg content and then translate the svg content to fit it into view box.

Please let me know if I missed anything.
Thanks